stable/concourse/templates/web-deployment.yaml

```diff
@@ -226,48 +161,24 @@ spec:
                 secretKeyRef:
                   name: {{ template "concourse.concourse.fullname" . }}
                   key: generic-oauth-client-secret
+            - name: CONCOURSE_GENERIC_OAUTH_DISPLAY_NAME
+              value: {{ default "" .Values.concourse.genericOauth.displayName | quote }}
             - name: CONCOURSE_GENERIC_OAUTH_AUTH_URL
-              valueFrom:
-                configMapKeyRef:
-                  name: {{ template "concourse.concourse.fullname" . }}
-                  key: generic-oauth-auth-url
+              value: {{ default "" .Values.concourse.genericOauth.authUrl | quote }}
             - name: CONCOURSE_GENERIC_OAUTH_AUTH_URL_PARAM
-              valueFrom:
-                configMapKeyRef:
-                  name: {{ template "concourse.concourse.fullname" . }}
-                  key: generic-oauth-auth-url-param
+              value: {{ default "" .Values.concourse.genericOauth.authUrlParam | quote }}
             - name: CONCOURSE_GENERIC_OAUTH_SCOPE
-              valueFrom:
-                configMapKeyRef:
-                  name: {{ template "concourse.concourse.fullname" . }}
-                  key: generic-oauth-scope
+              value: {{ default "" .Values.concourse.genericOauth.scope | quote }}
             - name: CONCOURSE_GENERIC_OAUTH_TOKEN_URL
-              valueFrom:
-                configMapKeyRef:
-                  name: {{ template "concourse.concourse.fullname" . }}
-                  key: generic-oauth-token-url
-            - name: CONCOURSE_TSA_HOST_KEY
-              value: "/concourse-keys/host_key"
-            - name: CONCOURSE_SESSION_SIGNING_KEY
-              value: "/concourse-keys/session_signing_key"
-            - name: CONCOURSE_TSA_AUTHORIZED_KEYS
-              value: "/concourse-keys/worker_key.pub"
-            {{- if .Values.credentialManager.enabled }}
+              value: {{ default "" .Values.concourse.genericOauth.tokenUrl | quote }}
+            {{- end }}
+            {{- if .Values.credentialManager.vault.enabled }}
             - name: CONCOURSE_VAULT_URL
-              valueFrom:
-                configMapKeyRef:
-                  name: {{ template "concourse.concourse.fullname" . }}
-                  key: vault-url
+              value: {{ default "" .Values.credentialManager.vault.url | quote }}
             - name: CONCOURSE_VAULT_PATH_PREFIX
-              valueFrom:
-                configMapKeyRef:
-                  name: {{ template "concourse.concourse.fullname" . }}
-                  key: vault-path-prefix
+              value: {{ default "/concourse" .Values.credentialManager.vault.pathPrefix | quote }}
             - name: CONCOURSE_VAULT_AUTH_BACKEND
-              valueFrom:
-                configMapKeyRef:
-                  name: {{ template "concourse.concourse.fullname" . }}
-                  key: vault-auth-backend
+              value: {{ default "" .Values.credentialManager.vault.authBackend | quote }}
```

Comment:
Maybe this should be required instead of an empty default to avoid undefined behavior?

Reply:
in the comments for `credentialManager.vault.authBackend`:

```
    ## vault authentication backend, leave this blank if using an initial periodic token
    ## currently supported backends: token, approle, cert.
    ##
    # authBackend:
```

Concourse seems to deal with env vars that are set as empty strings for other features, so unless we can demonstrate otherwise, I'd be hesitant to start wrapping everything in if statements. See also https://concourse.ci/creds.html#vault